### PR TITLE
Update submodules to versions with xproj

### PR DESCRIPTION
**Pull request by anurse**
We also have to update to react to some Logging changes.

/cc @victorhurdugaci @davidfowl @ChengTian

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@anurse__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Andrew Stanton-Nurse). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.Framework.ApplicationHost2/project.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     ],
     "exclude": [
         "../../submodules/NuGet.Versioning/src/NuGet.Versioning/compiler/**/*.cs",
-        "../../submodules/**/AssemblyInfo.cs"
+        "../../submodules/**/AssemblyInfo.cs",
+        "../../submodules/Logging/src/Microsoft.Framework.Logging/LoggingServiceCollectionExtensions.cs"
```
Comment (anurse) [inline src/Microsoft.Framework.ApplicationHost2/project.json]: If we don't exclude this, we have to pull DI Interfaces into the AppHost, which we don't really want to do.

File: src/Microsoft.Framework.ApplicationHost2/project.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
         "Newtonsoft.Json": "6.0.6",
         "Microsoft.Framework.Runtime.Roslyn.Interfaces": "1.0.0-*",
         "Microsoft.Framework.Runtime.Interfaces": "1.0.0-*",
-        "Microsoft.Framework.NotNullAttribute.Internal": { "type": "build", "version": "1.0.0-*" }
+        "Microsoft.Framework.NotNullAttribute.Internal": { "type": "build", "version": "1.0.0-*" },
```
Comment (victorhurdugaci) [inline src/Microsoft.Framework.ApplicationHost2/project.json]: Is this valid json?

Comment (anurse) [inline src/Microsoft.Framework.ApplicationHost2/project.json]: No, but it is valid JSON.NET :). Still, I'll fix it. It's an artifact of my initial solution which was to add DI.Interfaces :)

**Pull request merged**